python/cugraph/utilities/nx_factory.py

```diff
@@ -35,6 +36,11 @@ def convert_from_nx(nxG, weight=None):
         raise ValueError("nxG does not appear to be a NetworkX graph type")
 
     pdf = nx.to_pandas_edgelist(nxG)
+    if pdf["source"].dtype not in [np.int32, np.int64] or \
+            pdf["target"].dtype not in [np.int32, np.int64]:
+        pdf['source'] = pdf['source'].astype(str)
+        pdf['target'] = pdf['target'].astype(str)
```

Comment:
Can you add a comment here explaining why this needs to be done? It's not clear to the reader why this is important, why it can't be removed or changed, etc.